src/Radicle/Conversion.hs

```diff
@@ -43,22 +44,27 @@ typeMismatch expected actual =
     throwError $ "expected " <> expected <> ", encountered " <> name
   where
     name = case actual of
-             Rad.Atom _    -> "Atom"
-             Rad.Boolean _ -> "Boolean"
-             Rad.Dict _    -> "Dict"
-             Rad.Keyword _ -> "Keyword"
-             Rad.Lambda{}  -> "Lambda"
-             Rad.List _    -> "List"
-             Rad.Number _  -> "Number"
-             Rad.PrimFn _  -> "Primop"
-             Rad.Ref _     -> "Ref"
-             Rad.String _  -> "String"
-             Rad.Vec _     -> "Vec"
+        Rad.Atom{}       -> "Atom"
+        Rad.Boolean{}    -> "Boolean"
+        Rad.Dict{}       -> "Dict"
+        Rad.Keyword{}    -> "Keyword"
+        Rad.Lambda{}     -> "Lambda"
+        Rad.List{}       -> "List"
+        Rad.Number{}     -> "Number"
+        Rad.PrimFn{}     -> "Primop"
+        Rad.Ref{}        -> "Ref"
+        Rad.String{}     -> "String"
+        Rad.Vec{}        -> "Vec"
+        Rad.VState{}     -> "VState"
+        Rad.VEnv{}       -> "VEnv"
+        Rad.LambdaRec{}  -> "LambdaRec"
+        Rad.ProcHandle{} -> "ProcHandle"
+        Rad.Handle{}     -> "Handle"
 
-instance ToRad a => ToRad (Set a) where
+instance ToRad WithPos a => ToRad WithPos (Set a) where
     toRad = toRad . Set.toList
 
-instance (FromRad a, Ord a) => FromRad (Set a) where
+instance (FromRad WithPos a, Ord a) => FromRad WithPos (Set a) where
```

Comment:
I think `WithPos` can be replaced by a type variable.

Reply:
Here it can, but since we effectively always need `WithPos` I though that is less confusing